app/controllers/api/v1/statuses_controller.rb

```diff
@@ -61,7 +61,7 @@ def create
                                          status_params[:status],
                                          status_params[:in_reply_to_id].blank? ? nil : Status.find(status_params[:in_reply_to_id]),
                                          media_ids: status_params[:media_ids],
-                                         sensitive: status_params[:sensitive],
+                                         sensitive: status_params[:spoiler_text].blank? ? status_params[:sensitive] : true,
```

Comment:
Wouldn't it be easier to just have this line, without any others? Here you're already setting sensitive=true when spoiler_text=present. No reason to keep checking spoiler_text beyond this point.